CIP-0100/README.md

```diff
@@ -0,0 +1,298 @@
+
+</details>
+
+## Motivation
+
+With the advent of the Voltaire era, Cardano is moving towards a decentralized governance model. CIP-1694 addresses a potential technical implementation of ledger rules for creating, voting on, and ratifying proposed changes to the ledger. The ledger has no mechanism or desire to validate this metadata, and as a result, the official specification leaves the format of this metadata unspecified.
+
+To facilitate rich user experiences for the various governance actors, however, it would be beneficial to have a suggested universal format for this metadata, allowing deep and interconnected discovery and exploration of this metadata. This CIP seeks to provide that standard format, and a minimal set of fields for various governance actions, leaving the true depth of metadata to be defined later through the extensibility mechanism outlined below.
+
+While this specification is written with CIP-1694 in mind, many of the ideas should be equally suitable for any other governance proposal, provided that proposal has a mechanism for attaching metadata to a governance action.
+
+Explicitly, here are the goals this CIP is trying to satisfy:
+ - Standardize a format for rich metadata related to cardano governance
+ - Standardize a minimal and uncontroversial set of fields to support "Minimal Viable Governance"
+ - Leave that format open to extension and experimentation
+ - Enable tooling and ecosystem developers to build the best user experiences possible
+ - Provide a set of best practices that tooling and ecosystem developers can rely on for the health and integrity of this data
+
+## Specification
+
+This section outlines the high level format and requirements of this standard.
+
+      The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+      NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+      "OPTIONAL" in this document are to be interpreted as described in
+      [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
+      
+- On chain metadata actions in CIP-1694 (and likely any alternative proposals) have the notion of an "Anchor"; this is the URL and a hash for additional, non-ledger metadata about the action.
+- Tools which publish governance related transactions SHOULD publish metadata via these fields.
+- While that content MAY be in any format, following any standard or non-standard, for the remainder of this document SHOULD/MUST will refer to documents that are following this specification.
+- The content hosted at the anchor URL MUST be a JSON-LD document according to the rest of this specification.
+- This document SHOULD include `@context` and `@type` fields below to aid in interpretation of the document.  
+- The JSON document SHOULD be formatted for human readability, for the sake of anyone who is manually perusing the metadata.
+- That content SHOULD be hosted on a content addressable storage medium, such as IPFS or Arweave, to ensure immutability and long term archival.
```

Comment:
An issue with IPFS is that I believe links are always larger than the max size of metadata URL, that being 64 chars

Reply:
I was thinking that storing only the IPFS CID would make it fit; it appears to have a maximum length of 45 characters but I can't find any guarantee of this: https://github.com/multiformats/cid/issues/21

Reply:
The Algorand people were thinking of something like that: https://forum.algorand.org/t/asset-url-max-length-how-to-link-ipfs/3054/3

Reply:
Is the same restriction being applied to the `anchor` field? Keep in mind that anchors aren't being stored in the metadata of a transaction, so it might have different restrictions!

Reply:
Yes, this is a limit in the Ledger design... see CDDL [here](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/conway/impl/cddl-files/conway.cddl#L373C1-L373C25) technically 64 bytes I believe.

Reply:
We might want to ask for that to be raised to support some of the most reasonable likely urls; 64 bytes is pretty limiting!